src/main/java/frc/robot/GenericPID/Approximations/ApproximateDerivative.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.GenericPID.Approximations;
+
+import frc.robot.GenericPID.Debug;
+
+public class ApproximateDerivative {
+    private static final boolean debug = Debug.debug_Approximations;
+    //a sometimes static class which takes the derivative of a certain point of a function
+    //just use the derivative method, and a double function
+    //or track ongoing derivative by using nextDerivative()
+    //use either DoubleFunction unit or manual evaluation, either f or ylast
+    private double xlast;
+    private double ylast;
+    private DoubleFunction f;
+    //static instant version
+    public static double derivative(DoubleFunction f, double dx, double x) {
+        return (f.eval(x) - f.eval(x - dx)) / dx;
```

Comment:
Implement eval(double x) from DoubleFunction within this class (and also approximate integral), even if that means including a Motor and a GenericPID as parameters in the constructor.

Reply:
what??